### DISCUSSION TITLE
Longer Audios

### DISCUSSION THREAD

**Discussion by guiaugustinegri** [souzatharsis/podcastfy#168] [Q&A]
What is the relationship between the size of the content sent to podcastify and the "word_count" setting to get longer audios? I tested several settings but the maximum I could get was 4 minutes.

**Comment by souzatharsis** [comment]
LLMs are known for their limited output number of tokes size.
Long form podcast generation has become an issue of the highest priority since multiple requests have been made.

https://github.com/souzatharsis/podcastfy/issues/166

I'll work on it next; No clue what the solution will be since there isn't any known best practice for it but it will be fun to develop a robust solution for it.

stay tuned

**Reply by guiaugustinegri** [reply]
Thanks for the answer. Today, the LLM that generates the podcast script is Gemini, right? What I do to get larger responses in LLMS is to use the OpenAI API, I make the request using the gpt4-o model and create a thread, which is a native option. This way, in addition to having the initial context, it can remember previous requests for each request, working similarly to the chatgpt app. So I ask it to send me the responses in separate blocks, so I can receive more tokens. I believe this could be an option to be tested.

**Comment by souzatharsis** [comment]
As of a couple of days ago you can choose 100+ llm models to generate the
transcript including gpt4-o.
however, in my experience, llms are trained to give the shortest answer
possible
i'm currently working on a new multi-prompt transcript generation approach
leveraging a content chunking with contextual linking approach
In that way, we generate detailed transcripts per input chunck and then
stitch responses together while maintaining cohesion.

Easier said than done, but I am almost there.

Would love your feedback once done.

Best Regards,

--
Thársis
souzatharsis.com <http://www.souzatharsis.com/>
<http://linkedin.com/in/tharsissouza>

**Reply by guiaugustinegri** [reply]
Excellent! It's going to be awesome =)

**Comment by souzatharsis** [comment]
I've now managed to implement long-form podcast generation. Please see sample below. 
Abstracting away the TTS model used (OpenAI), would love feedback on the long-form transcript generated!

**Sample 1**: open-notebook.ai and its sub-pages
TTS Backend: OpenAI
- Audio: https://drive.google.com/file/d/1isp2gnfjnf-tyhgi_FREIbf02EafYjK4/view?usp=drive_link
- Transcript: https://drive.google.com/file/d/1XMbD7bZtwqN5Hs-2ysmxeeNO2s6xMXbW/view?usp=sharing

**Sample 2**: Some tech docs on surfsense.net shared by the author [here](https://justpaste.it/ho3go)
TTS Backend: OpenAI
- Audio: https://drive.google.com/file/d/1MyLF6RcLg3o7lp1bZA9ZStO0oocdMIhB/view?usp=drive_link
- Transcript: https://drive.google.com/file/d/17nhnPpf_r3QzQ4Kq-8pgwQ_P6hYm_LfS/view?usp=drive_link

**Sample 3**: Transformer pdf paper
TTS Backend: OpenAI
- Audio: https://drive.google.com/file/d/1iYgMjqw7ct8yVCWS4oACQWEZJmsocYpT/view?usp=drive_link
- Transcript: https://drive.google.com/file/d/1LqCL26e_BtwuLzCE-eFzVGiisWCxknrz/view?usp=drive_link

**Reply by MODSetter** [reply]
Hi according to my tests previous version failed to catch the small changelog content but it does now in the end.

**Reply by souzatharsis** [reply]
@MODSetter also replied with "Amazing it even captured git commits XD"

**Reply by MalikMAlna** [reply]
Listening to Sample 3 right now. 

So far it sounds pretty good! Will have more to say sometime within the next 24 hours after digging through it and the other samples in reference to the original transcript.

**EDIT:** Quick heads up, noticed some repetition between both AI hosts saying the same thing around 13:00 for Sample 3. Feels like that probably wasn’t intentional based on the context of what they were talking about earlier.

Otherwise, didn't encounter any other immediately obvious issues.

**Reply by ralyodio** [reply]
Is this merged?

**Reply by ralyodio** [reply]
How do I specify length of audio?

**Reply by KlaasYntema** [reply]
I listened to all three of them and besides the feedback MalikMAlna provided, they sound pretty good!

**Reply by souzatharsis** [reply]
Thanks for the feedback! @KlaasYntema @MODSetter @MalikMAlna !

@ralyodio , it's not merged yet. I'll merge it once I collect enough feedback.

**Reply by toilv** [reply]
I still only get 7 minutes for file. geminimulti with --longform . Any suggesttion? I need file length at least 20 mins.

**Comment by souzatharsis** [comment]
Sample 4: Gemini Prompt Engineering Guide
TTS Backend: ElevenLabs

Transcript: https://drive.google.com/file/d/1ed2Xl9zPLyXPQNqJEcxGedgp9cMyN97T/view?usp=drive_link
Audio: https://drive.google.com/file/d/1jdnZC_3R0A9TNt9IPYZORz8_oT_o3YaX/view?usp=drive_link

**Comment by souzatharsis** [comment]
This has been implemented and released - I'd love feedback!

See https://github.com/souzatharsis/podcastfy/blob/main/usage/how-to.md#how-to-generate-longform-podcasts

## [0.3.6] - 2024-11-13

### Added
- Add longform podcast generation support
  - Users can now generate longer podcasts (20-30+ minutes) using the `--longform` flag in CLI or `longform=True` in Python API
  - Implements "Content Chunking with Contextual Linking" technique for coherent long-form content
  - Configurable via `max_num_chunks` and `min_chunk_size` parameters in conversation config
  - `word_count` parameter removed from conversation config as it's no longer used

**Reply by toilv** [reply]
I still only get 7 minutes for file. geminimulti with --longform . Any suggesttion? I need file length at least 20 mins.

**Comment by ralyodio** [comment]
I want at least an hour.
